tools/install.py

```diff
@@ -192,27 +196,41 @@ def run(args):
   variables = conf['variables']
   target_defaults = conf['target_defaults']
 
-  # argv[2] is a custom install prefix for packagers (think DESTDIR)
-  # argv[3] is a custom install prefix (think PREFIX)
+  # args[1] is a custom install prefix for packagers (think DESTDIR)
+  # args[2] is a custom install prefix (think PREFIX)
   # Difference is that dst_dir won't be included in shebang lines etc.
-  dst_dir = args[2] if len(args) > 2 else ''
+  dst_dir = args[1] if len(args) > 1 else ''
 
-  if len(args) > 3:
-    node_prefix = args[3]
+  if len(args) > 2:
+    node_prefix = args[2]
 
   # install_path thus becomes the base target directory.
   install_path = dst_dir + node_prefix + '/'
 
-  cmd = args[1] if len(args) > 1 else 'install'
+  cmd = args[0].lower() if len(args) else 'install'
 
-  if os.environ.get('HEADERS_ONLY'):
-    if cmd == 'install': return headers(install)
-    if cmd == 'uninstall': return headers(uninstall)
-  else:
-    if cmd == 'install': return files(install)
-    if cmd == 'uninstall': return files(uninstall)
+  if cmd not in ('install', 'uninstall'):
+    raise RuntimeError('Bad command: %s\n' % cmd)
+
+  for option, value in opts:
+    if option in ('-c', '--components'):
+      value = (value.lower() or 'node,npm,headers').split(',')
+      components.intersection_update([v for v in value if v in components])
+
+  # Even if NPM is not asked to install, if "node_install_npm" is set, then
+  # install npm_files anyway, unless it is 'headers' installation
+  if 'headers' not in components and 'true' == variables.get('node_install_npm'):
+    components.update(['npm'])
+
+  files(components, install if cmd == 'install' else uninstall)
 
-  raise RuntimeError('Bad command: %s\n' % cmd)
 
 if __name__ == '__main__':
-  run(sys.argv[:])
+
+  try:
+    opts, args = getopt.getopt(sys.argv[1:], 'c:', ['components='])
```

Comment:
Using `getopt` is fine IMO but I'll just note that we use `optparse` everywhere else.